### PR TITLE
feature/create_QuestionPage_UI

### PR DESCRIPTION
**Pull request by AnaIoramashvili**

### CODE REVIEW

File: QuizApp/Screens/QuestionPage/AnswerCell.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        view.backgroundColor = Constants.Colors.neutralLighterGray
+        view.layer.cornerRadius = Constants.AnswerCellConstants.containerCornerRadius
+        view.clipsToBounds = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    private let titleLabel: UILabel = {
+        let label = UILabel()
+        label.textAlignment = .left
```
Comment (tamokakhidze) [inline QuizApp/Screens/QuestionPage/AnswerCell.swift]: this is default alignment I think

File: QuizApp/Screens/QuestionPage/AnswerCell.swift [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    private func setupView() {
+        selectionStyle = .none
+        backgroundColor = .clear
+    }
+    
+    private func setUpHierarchy() {
+        contentView.addSubview(containerView)
+        containerView.addSubview(titleLabel)
+        containerView.addSubview(scoreLabel)
+        containerView.addSubview(starImageView)
```
Comment (tamokakhidze) [inline QuizApp/Screens/QuestionPage/AnswerCell.swift]:
```suggestion
        containerView.addSubviews(
        titleLabel, 
        scoreLabel, 
        starImageView
        )
```

**Review by tamokakhidze — APPROVED** [review]

**Review by LukaGujejiani — APPROVED** [review]

File: QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        return label
+    }()
+    
+    private let starImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: Constants.AnswerCellConstants.whiteStarImage)
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.isHidden = true
+        return imageView
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift]: You do not need this if you will do score label with attributed string

File: QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift:1]: use stack view for components

File: QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        if isCorrect {
+            // Correct answer
+            containerView.backgroundColor = Constants.AnswerCellConstants.success
+            titleLabel.textColor = Constants.Colors.neutralWhite
+            scoreLabel.isHidden = false
+            starImageView.isHidden = false
+        } else {
+            // Wrong answer
+            containerView.backgroundColor = Constants.AnswerCellConstants.wrong
+            titleLabel.textColor = Constants.Colors.neutralWhite
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift:130]: why do you need to set textcolor again, you are not changing the colour though

File: QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            scoreLabel.isHidden = true
+            starImageView.isHidden = true
+        }
+    }
+    
+    func markAsCorrect() {
+        // Show correct answer cell styling without +1 or star
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift]: better to delete the comment or write this above the function

File: QuizApp/Screens/QuestionPage/QuizViewController.swift [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        let tableView = UITableView()
+        tableView.backgroundColor = .clear
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(
+            AnswerCell.self,
+            forCellReuseIdentifier: AnswerCell.identifier
+        )
+        tableView.separatorStyle = .none
+        tableView.isScrollEnabled = false
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/QuizViewController.swift]: why you are setting tableView.isScrollEnabled = false, what if you have more items

File: QuizApp/Screens/QuestionPage/QuizViewController.swift [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        headerView.configure(with: "პროგრამირება")
+        progressView.updateProgress(currentQuestion: 1, totalQuestions: 10)
+        progressView.updatePoints(points: 1)
+        questionView.configure(question: "რომელია ყველაზე პოპულარული პროგრამირების ენა?")
+    }
+    
+    // Sample data
+    private var answers = ["Python", "Java", "C++", "Kotlin"]
+    private var selectedAnswerIndex: Int?
+    private var correctAnswerIndex = 1
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/QuizViewController.swift:154]: I guess this is just for testing

File: QuizApp/Screens/QuestionPage/Views/QuizProgressView.swift [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        return label
+    }()
+    
+    private let starImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: Constants.QuizProgressViewConstants.starImage)
+        imageView.tintColor = Constants.Colors.yellowPrimary
+        imageView.contentMode = .scaleAspectFit
+        return imageView
+    }()
```
Comment (AniDzamelashvili) [inline QuizApp/Screens/QuestionPage/Views/QuizProgressView.swift]: Same, here you can have attributed string instead of starImage

**Review by AniDzamelashvili — APPROVED** [review]

File: QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift (hunk 4 above)
Comment (AnaIoramashvili) [reply QuizApp/Screens/QuestionPage/Cell/AnswerCell.swift:130]: from black color it changes to white color after you select the cell.

File: QuizApp/Screens/QuestionPage/QuizViewController.swift (hunk 7 above)
Comment (AnaIoramashvili) [reply QuizApp/Screens/QuestionPage/QuizViewController.swift:154]: yes it is

**Pull request merged**